engine/src/test/java/io/zeebe/engine/processing/bpmn/activity/CallActivityTest.java

```diff
@@ -222,6 +222,34 @@ public void shouldPropagateVariablesToParent() {
         .containsExactly(tuple(workflowInstanceKey, "y", "2"));
   }
 
+  @Test
+  public void shouldNotPropagateVariablesToParentIfDisabled() {
```

Comment:
Would be good to add a test case for the following cases:
- `propagateAllChildVariables = false` && no output mappings
- `propagateAllChildVariables = false` && output mappings
- `propagateAllChildVariables = true` && no output mappings
- `propagateAllChildVariables = true` && output mappings

Reply:
The cases are almost covered by the following tests: 

- `propagateAllChildVariables = false` && no output mappings -> `shouldNotPropagateVariablesToParentIfDisabled`
- `propagateAllChildVariables = false` && output mappings -> behaves like  `shouldApplyOutputMappings` but we can add an additional test
- `propagateAllChildVariables = true` && no output mappings -> `shouldPropagateVariablesToParent`  
- `propagateAllChildVariables = true` && output mappings -> `shouldApplyOutputMappings`

Reply:
Sorry, I did not look at the existing tests only at new code. I think it is fine like this, although I'm also fine if you add the "propagateAllChildVariables = false && output mappings" case explicitly.